File: python/mathfunc3.py
import os, sys, time

# Constants for the block characters used to represent pixels:
TOP_BLOCK    = chr(9600)
BOTTOM_BLOCK = chr(9604)
FULL_BLOCK   = chr(9608)

DELAY = 0.05
FUNC = eval('lambda x, y: (x | y) % 7')


y = 0
while True:
    width = os.get_terminal_size()[0] - 2
    for x in range(width):
        topBit = FUNC(x, y)
        bottomBit = FUNC(x, y + 1)

        # Patterns often look better if we use True for the black pixels:
        topBit = not topBit
        bottomBit = not bottomBit

        if topBit and bottomBit:
            print(FULL_BLOCK, end='')
        elif topBit and not bottomBit:
            print(TOP_BLOCK, end='')
        elif not topBit and bottomBit:
            print(BOTTOM_BLOCK, end='')
        else:
            print(' ', end='')
    print(flush=True)

    y += 2
    time.sleep(DELAY)

File: typescript/mathfunc3.ts
// Math Func scroll art, by Al Sweigart al@inventwithpython.com
// To get ts types, run: npm i --save-dev @types/readline

namespace MathFunc {
    const DELAY: number = 100;
    //const FUNC = (x: number, y: number): boolean => ((x ^ y) % 5) !== 0;
    //const FUNC = (x: number, y: number): boolean => ((x & y) & (x ^ y) % 19) !== 0;
    //const FUNC = (x: number, y: number): boolean => ((x * 64) % y) !== 0;
    const FUNC = (x: number, y: number): boolean => ((x | y) % 7) !== 0;

    const TOP_BLOCK: string = String.fromCodePoint(9600);
    const BOTTOM_BLOCK: string = String.fromCodePoint(9604);
    const FULL_BLOCK: string = String.fromCodePoint(9608);
    const EMPTY_BLOCK: string = ' ';

    async function sleep(milliseconds: number): Promise<void> {
        return new Promise(resolve => setTimeout(resolve, milliseconds));
    }

    process.on('SIGINT', function() {
        console.log("\nMath Func by Al Sweigart al@inventwithpython.com");
        process.exit();
    });

    function getTerminalWidth(): number {
        if (process.stdout.isTTY) {
            return process.stdout.columns - 1; // -1 is because Windows puts a newline when printing to the last column
        } else {
            return 80;
        }
    }

    async function main(): Promise<void> {
        let y: number = 0;
        while (true) {
            let width = getTerminalWidth() - 1;
            let line: string = '';
            for (let x = 0; x < width; x++) {
                let topBit: boolean = FUNC(x, y);
                let bottomBit: boolean = FUNC(x, y + 1);

                // Flipping the bits because I think they often look better this way:
                topBit = !topBit;
                bottomBit = !bottomBit;

                if (topBit && bottomBit) {
                    line += FULL_BLOCK;
                } else if (topBit && !bottomBit) {
                    line += TOP_BLOCK;
                } else if (!topBit && bottomBit) {
                    line += BOTTOM_BLOCK;
                } else {
                    line += ' ';
                }
            }
            console.log(line);
            y += 2;
            await sleep(DELAY);
        }
    }

    main();
}